app/controllers/application_controller.rb

```diff
@@ -397,6 +397,16 @@ def parameter_filter_context
     Foreman::ParameterFilter::Context.new(:ui, controller_name, params[:action])
   end
 
+  def wrap_for_virt_column_select(base_scope)
+    virt_column = params[:order]&.split(' ')&.first
+    select_method = "select_#{virt_column}"
+    if virt_column.blank? || model_of_controller.columns_hash[virt_column] || !base_scope.respond_to?(select_method)
+      base_scope.search_for(params[:search], :order => params[:order]).paginate(:page => params[:page], :per_page => params[:per_page])
+    else
+      base_scope.send(select_method).search_for(params[:search]).order(params[:order]).paginate(:page => params[:page], :per_page => params[:per_page])
```

Comment:
> UI does not accept per_page being set to all?

Nope, otherwise we'd have a button suggesting that.

> Also, thinking out loud, from the name I wouldn't really expect it to deal with pagination and so on, but that might be just me

Naming is hard, any suggestion is more than welcome :cookie:

Reply:
Does this break per_page=all for api?

Reply:
> Does this break per_page=all for api?

`wrap_for_virt_column_select` method is only for UI controllers, thus no.

API controllers most (if not all) of the time use `resource_scope_for_index` method, which is overridden in this PR:
https://github.com/theforeman/foreman/pull/10381/files#diff-f97adc72a387ed049f1b789f93f9d5c592f481e9e37c6fb75eabfcc2f36d467eR172

This override should not affect `per_page=all`, since it's either uses `super` to use old behavior or explicitly deals with it in https://github.com/theforeman/foreman/pull/10381/files#diff-f97adc72a387ed049f1b789f93f9d5c592f481e9e37c6fb75eabfcc2f36d467eR181.